content/en/blog/_posts/2022-08-25-pod-security/index.md

```diff
@@ -0,0 +1,90 @@
+date: 2022-08-25
+slug: pod-security-admission-stable
+canonicalUrl: https://kubernetes.dev/blog/2022/08/25/pod-security-admission-stable/
+---
+
+ **Authors:** Tim Allclair and Sam Stoelinga (Google)
+
+The release of Kubernetes v1.25 marks a major milestone for Kubernetes out-of-the-box pod security
+controls: Pod Security admission (PSA) graduated to stable, and Pod Security Policy (PSP) has been
+removed.
+[PSP has been deprecated since Kubernetes v1.21](https://kubernetes.io/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/),
+but as of v1.25 it will no longer be functional.
+
+Pod Security Admission was designed as PSP's replacement, and makes it easier to enforce predefined
+[Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/) by
+simply adding a label to a namespace. The Pod Security Standards are maintained by the K8s
+community, which means you automatically get updated security policies when new K8s features are
+introduced.
+
+
+## What’s new since Beta?
+
+Pod Security Admission hasn’t changed much since Beta (in Kubernetes v1.23). The focus has been on
+improving the user experience, while continuing to maintain a high quality and scalability bar.
+
+**Improved violation messages:**
+
+Minor improvements to violation messages were made to produce
+[fewer duplicate messages](https://github.com/kubernetes/kubernetes/pull/107698), so instead of
+getting this (since the baseline & restricted policies have overlapping capabilities checks):
+
+```
+pods "admin-pod" is forbidden: violates PodSecurity "restricted:latest": non-default capabilities (container "admin" must not include "SYS_ADMIN" in securityContext.capabilities.add), unrestricted capabilities (container "admin" must not include "SYS_ADMIN" in securityContext.capabilities.add)
+```
+
+You get:
+
+```
+pods "admin-pod" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "admin" must not include "SYS_ADMIN" in securityContext.capabilities.add)
+```
+
+**Improved namespace warnings:**
+
+When modifying the `enforce` Pod Security labels on a namespace, PSA checks all existing pods for
+violations and surfaces warning if any are out of compliance. Now,
+[warnings are aggregated](https://github.com/kubernetes/kubernetes/pull/105889) for pods with
+identical violations, making large namespaces with many replicas much more manageable. For example:
+
+```
+Warning: frontend-h23gf2: allowPrivilegeEscalation != false
+Warning: myjob-g342hj (and 6 other pods): host namespaces, allowPrivilegeEscalation != false Warning: backend-j23h42 (and 1 other pod): non-default capabilities, unrestricted capabilities
+```
+
+Additionally, when applying a non-privileged to a namespace that has been
+[configured to be exempt](https://kubernetes.io/docs/concepts/security/pod-security-admission/#exemptions),
+you will now get a warning alerting you to this fact:
```

Comment:
```suggestion
Additionally, when you deploy a non-privileged Pod to a namespace that you
[configured to be exempt](https://kubernetes.io/docs/concepts/security/pod-security-admission/#exemptions),
you will now get a warning alerting you to this fact:
```

Reply:
This applies to the action of labeling a namespace (the word `label` was missing). How about:

```suggestion
Additionally, when you apply a non-privileged label to a namespace that has been
[configured to be exempt](https://kubernetes.io/docs/concepts/security/pod-security-admission/#exemptions),
you will now get a warning alerting you to this fact:
```

I prefer `has been configured ...` to `you configured ...` since the configuration may have been applied by someone else (e.g. GKE cluster provisioning).